extra_geom/detectors.py

```diff
@@ -1810,13 +1853,34 @@ class Epix100Geometry(EpixGeometryBase):
     detector_type_name = 'ePix100'
     pixel_size = 50e-6
     inner_pixel_size = 175e-6
-    asic_gap = inner_pixel_size / pixel_size - 1
+    asic_gap = 2 * (inner_pixel_size - pixel_size) / pixel_size
+    # ePix100 should have 352 pixel per row, but the data we store has 354
     frag_ss_pixels = 352  # rows
     frag_fs_pixels = 384  # columns
     expected_data_shape = (
         EpixGeometryBase.n_modules,
         2 * frag_ss_pixels,
-        2 * frag_fs_pixels)
+        2 * frag_fs_pixels
+    )
+
+    @classmethod
+    def _ensure_shape(cls, data):
+        """Ensure image data has the proper shape.
+
+        As a ePix frame is read out and saved as a single array, the
+        public interface of this geometry implementation supports
+        automatic reshaping to adding the modules dimension.
+        """
+        if data.shape[-2:] == (2 * cls.frag_ss_pixels + 4, 2 * cls.frag_fs_pixels):
+            # EuXFEL data stored extra 2 row per ASIC used for diagnostic
+            #   - pixel max (the first and last rows in the pixel array)
+            #   - baseline (next rows)
+            # we removed them here as they do not contain data
+            data = np.concatenate((data[..., 1:cls.frag_ss_pixels+1, :],
+                                   data[..., -1-cls.frag_ss_pixels:-1, :]),
+                                  axis=-2)
```

Comment:
```suggestion
            data = data[...,2:-2,:]
```
There are two diagnostic rows on one edge of each ASIC. But ASICs look at each other in the data. Therefore, it turns out that these are the first two and the last two rows in the original data.